azkaban-exec-server/src/main/java/azkaban/dag/DagService.java

```diff
@@ -81,28 +87,14 @@ void killDag(final Dag dag) {
 
   /**
    * Shuts down the service and waits for the tasks to finish.
-   *
-   * <p>Adopted from
-   * <a href="https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html">
-   *   the Oracle JAVA Documentation.
-   * </a>
    */
-  void shutdownAndAwaitTermination() {
-    this.executorService.shutdown(); // Disable new tasks from being submitted
-    try {
-      // Wait a while for existing tasks to terminate
-      if (!this.executorService.awaitTermination(SHUTDOWN_WAIT_TIMEOUT, TimeUnit.SECONDS)) {
-        this.executorService.shutdownNow(); // Cancel currently executing tasks
-        // Wait a while for tasks to respond to being cancelled
-        if (!this.executorService.awaitTermination(SHUTDOWN_WAIT_TIMEOUT, TimeUnit.SECONDS)) {
-          logger.error("The DagService did not terminate.");
-        }
-      }
-    } catch (final InterruptedException ie) {
-      // (Re-)Cancel if current thread also interrupted
-      this.executorService.shutdownNow();
-      // Preserve interrupt status
-      Thread.currentThread().interrupt();
-    }
+  void shutdownAndAwaitTermination() throws InterruptedException {
```

Comment:
Just one more question. Where will you call the method? I don't see any callers.

Reply:
Thanks.

The class is not used in production code yet. But you can see example usages in the unit tests.